Move the shebang of the vendored qatlib and QATzip `autogen.sh` scripts
above their license headers.

With the shebang not on the first line, running `./autogen.sh` fails with
`exec format error`, and `patchShebangs` does not rewrite it.

Fixed upstream in qatlib and QATzip:
* https://github.com/intel/qatlib/issues/128
* https://github.com/intel/QATzip/issues/143
and bumped in Ceph with https://github.com/ceph/ceph/pull/72402
(backport pending in https://github.com/ceph/ceph/pull/73061).

--- a/src/qatlib/autogen.sh
+++ b/src/qatlib/autogen.sh
@@ -1,3 +1,4 @@
+#!/bin/sh
 #################################################################
 #
 #   BSD LICENSE
@@ -33,6 +34,5 @@
 # 
 #
 #################################################################
-#!/bin/sh
 
 autoreconf -if
--- a/src/qatzip/autogen.sh
+++ b/src/qatzip/autogen.sh
@@ -1,3 +1,4 @@
+#!/bin/sh
 ################################################################
 #   BSD LICENSE
 #
@@ -31,6 +32,5 @@
 #   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 ################################################################
-#!/bin/sh
 mkdir -p m4
 autoreconf -vif
